kube_hunter/modules/discovery/hosts.py

```diff
@@ -61,12 +61,33 @@ def __init__(self, pod=False, active=False, predefined_hosts=None):
 class HostDiscoveryHelpers:
     # generator, generating a subnet by given a cidr
     @staticmethod
-    def generate_subnet(ip, sn="24"):
+    def generate_subnet(ip, sn="24", ignore=None):
         logger.debug(f"HostDiscoveryHelpers.generate_subnet {ip}/{sn}")
         subnet = f"{ip}/{sn}"
         for ip in IPNetwork(subnet):
-            logger.debug(f"HostDiscoveryHelpers.generate_subnet yielding {ip}")
-            yield ip
+            if ignore and any(ip in s for s in ignore):
+                logger.debug(f"HostDiscoveryHelpers.generate_subnet DENIED {ip}")
+            else:
+                logger.debug(f"HostDiscoveryHelpers.generate_subnet yielding {ip}")
+                yield ip
+
+    @staticmethod
+    def gen_ips():
+        ignore = list()
+        scan = list()
+        for cidr in config.cidr:
+            try:
+                ip, sn = cidr.split("/")
+            except ValueError:
+                logger.exception(f'Unable to parse CIDR "{config.cidr}"')
+                break
+            if ip.startswith("!"):
+                ignore.append(IPNetwork(f"{ip[1:]}/{sn}"))
```

Comment:
You previously parsed `cidr` into `ip` and `sn` and then joined them here.
You could just pass the cidr variable to `IPNetwork` (which will also consider a single IP as /32 subner)
```suggestion
            if cidr.startswith("!"):
                ignore.append(IPNetwork(cidr[1:]))
```